### PR TITLE
[dashboard] fix dashboard conftest visibility issue

### PR DESCRIPTION
**Pull request by aslonnie**
addresses 1 issue in https://github.com/ray-project/ray/pull/36769 where the rule cannot build due to visibility error.

### CODE REVIEW

File: python/ray/tests/BUILD [hunk 1]
```diff
@@ -4,6 +4,7 @@ py_library(
   name = "conftest",
   srcs = glob(["**/conftest.py"]),
   visibility = [
+    "//dashboard:__pkg__",
```
Comment (aslonnie) [inline python/ray/tests/BUILD:7]: part of the dashboard symlink madness I guess.

**Comment by can-anyscale** [comment]
What tests pass after this fix? There are some python test failures in CI as well.

**Comment by aslonnie** [comment]
> What tests pass after this fix?

see
https://github.com/ray-project/ray/pull/36769

this is one of the errors when trying to run:

bezel run @hedron_compile_commands//:refresh_all

the bazel rule in dashboard does not build, because the visibility is set for the symlink but not for the real directory.

**Comment by aslonnie** [comment]
> There are some python test failures in CI as well.

not 100% sure, but I think they are unrelated.

**Review by can-anyscale — APPROVED** [review]

**Review by raulchen — APPROVED** [review]

**Pull request merged**